File: 15-Java8/src/by/gsu/epamlab/beans/StrongTrial.java
package by.gsu.epamlab.beans;

public class StrongTrial extends Trial{
    private final static int DIVIDER = 2;

    public StrongTrial() {
    }

    public StrongTrial(String name, int mark1, int mark2) {
        super(name, mark1, mark2);
    }

    @Override
    public Trial getCopy(){
        return new StrongTrial(getName(), getMark1(), getMark2());
    }

    @Override
    public boolean isPassed() {
        return getMark1() / DIVIDER + getMark2() >= Trial.PASS_MARK;
    }
}

File: 22-jsOop/beans/StrongTrial.js
const Trial = require("./Trial.js");

class StrongTrial extends Trial{
    static DIVIDER = 2;

    isPassed() {
        return this._mark1 / StrongTrial.DIVIDER + this._mark2 >= Trial.PASS_MARK;
    }

    getCopy(){
        return new StrongTrial(this._account, this._mark1, this._mark2);
    }
}
module.exports = StrongTrial;